lablup/backend.ai
# feature: add new user's initial grace period for idle cheker

How to set for testing
1. Save the file below as <FILE_NAME>
```
{
    "app-streaming-packet-timeout": "",
    "checkers": {
        "network_timeout": {
            "threshold": "500"
        },
        "timeout": {
            "threshold": "500"
        },
        "user_grace_period": {
            "user_initial_grace_period": "1200"
        },
        "utilization": {
            "initial-grace-period": "120",
            "resource-thresholds": {
                "cpu_util": {
                    "average": "10"
                },
                "cuda_mem": {
                    "average": "10"
                },
                "cuda_util": {
                    "average": "10"
                },
                "mem": {
                    "average": "10"
                }
            },
            "thresholds-check-operator": "or",
            "time-window": "15"
        }
    },
    "enabled": "network_timeout"
}
```
2. Put the file to etcd
```
./backend.ai mgr etcd put-json config/idle <FILE_NAME>
```

---
The return value is like below.
```
{
   "network_timeout": {
      "remaining": 20.11,
      "remaining_time_type":"expire_after",
      "extra":null
   },
   "session_lifetime":{
      "remaining":null,
      "remaining_time_type":"expire_after",
      "extra":null
   },
   "utilization":{
      "remaining":-1.0,
      "remaining_time_type":"grace_period",
      "extra":{
         "cpu_util":[1796.4614285714288, 10.0],
         "mem":[43, 10],
         "cuda_util":[0.0, 10.0],
         "cuda_mem":[0, 10]
      }
   }
}
```
